server/tox.ini

```
@@ -55,3 +56,17 @@ set_env =
     PYTHONPATH = {toxinidir}/charm/src:{toxinidir}/charm/lib
 commands =
     pytest --cov=charm/src charm/tests/unit
+
+[testenv:openapi-check]
+description = Check server OpenAPI spec is up-to-date
+allowlist_externals = bash
+set_env =
+    FLASK_APP = devel.openapi_app
+commands =
+    bash -c 'if ! diff -u <(flask spec | python -m json.tool --sort-keys) <(python -m json.tool --sort-keys ./openapi.json) > /dev/null; then \
+        echo >&2 ""; \
+        echo >&2 "ERROR: OpenAPI spec is out of date."; \
+        echo >&2 "To update it, run:"; \
+        echo >&2 "  FLASK_APP=devel.openapi_app flask spec --output ./openapi.json"; \
```

I think we should refer to use uv directly instead, what do you think?
```suggestion
        echo >&2 "  FLASK_APP=devel.openapi_app uv run flask spec --output ./openapi.json --quiet"; \
```